hawtio-system/src/test/java/io/hawt/util/StringsTest.java

```diff
@@ -43,6 +44,56 @@ public void split() {
         }
     }
 
+    @Nested
+    class PropertiesTest {
+
+        @Test
+        public void oneLevelSystemProperties() {
+            System.setProperty("hawtio.what", "world");
+            assertThat(Strings.resolvePlaceholders(null), nullValue());
+            assertThat(Strings.resolvePlaceholders("Hello"), is("Hello"));
+            assertThat(Strings.resolvePlaceholders("Hello ${hawtio.what}!"), is("Hello world!"));
+            assertThat(Strings.resolvePlaceholders("${hawtio.what}"), is("world"));
+            assertThat(Strings.resolvePlaceholders("${hawtio.what}, hello ${hawtio.what}!"), is("world, hello world!"));
+        }
+
+        @Test
+        public void valuesWithPlaceholdersAreNotResolved() {
+            System.setProperty("hawtio.what", "${world}");
+            assertThat(Strings.resolvePlaceholders("Hello ${hawtio.what}!"), is("Hello ${world}!"));
```

Comment:
there seems to be no test that verifies `Hello ${world}` resolves to `Hello ${world}` (a case when the placeholder system property doesn't exist in the first place).